src/front/wgsl/mod.rs

```diff
@@ -296,12 +297,20 @@ impl<'a> Error<'a> {
                 labels: vec![(bad_span.clone(), "expected unsigned integer".into())],
                 notes: vec![],
             },
-
             Error::BadScalarWidth(ref bad_span, width) => ParseError {
                 message: format!("invalid width of `{}` bits for literal", width as u32 * 8,),
                 labels: vec![(bad_span.clone(), "invalid width".into())],
                 notes: vec!["the only valid width is 32 for now".to_string()],
             },
+            Error::BadMatrixScalarKind(
+                ref span,
+                kind,
+                width,
+             ) => ParseError {
+                message: format!("matrix scalar type must be `f32`, but found `{}`", kind.to_wgsl(width)),
+                labels: vec![(span.clone(), "must be `f32`".into())],
```

Comment:
Let's only mention the kind here (i.e. floating-point) since f64 might be allowed as well and once we support WGSL's f16 extension that would as well.